### ISSUE TITLE
When I set this training parameter, "overlap_mask=False",  I got this error. IndexError: The shape of the mask [66, 1] at index 0 does not match the shape of the indexed tensor [16, 160, 160] at index 0

### ISSUE THREAD

**Issue by saka3324** [ultralytics/ultralytics#585]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

When I train without the overlap_mask parameter, it works fine.
This line is causing the error.
```
if self.overlap:
  gt_mask = torch.where(masks[[i]] == mask_idx.view(-1, 1, 1), 1.0, 0.0)
else:
  gt_mask = masks[batch_idx == i][mask_idx]
```
shape of [66, 1] and [16,160,160] are completely different value.
I don't know how to solve this.

Thank you.
Log
```
yolo\engine\trainer: task=segment, mode=train, model=yolov8s-seg.pt, data=data.yaml, epochs=200, patience=50, batch=16, 
imgsz=640, save=True, cache=False, device=, workers=8, project=None, name=None, exist_ok=False, pretrained=False, 
optimizer=SGD, verbose=False, seed=0, deterministic=True, single_cls=False, image_weights=False, rect=False, cos_lr=False, 
close_mosaic=10, resume=False, overlap_mask=False, mask_ratio=4, dropout=0.0, val=True, save_json=False, save_hybrid=False, 
conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, 
save_conf=False, save_crop=False, hide_labels=False, hide_conf=False, vid_stride=1, line_thickness=3, visualize=False, 
augment=False, agnostic_nms=False, retina_masks=False, format=torchscript, keras=False, optimize=False, int8=False, 
dynamic=False, simplify=False, opset=17, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, 
warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, fl_gamma=0.0, 
label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, 
perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, hydra={'output_subdir': None, 'run': 
{'dir': '.'}}, v5loader=False, save_dir=runs\segment\train
Overriding model.yaml nc=80 with nc=2

Traceback (most recent call last):
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\Scripts\yolo.exe\__main__.py", line 7, in <module>
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\yolo\cli.py", line 141, in entrypoint
    cli(cfg)
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\yolo\cli.py", line 82, in cli
    func(cfg)
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\site-packages\hydra\main.py", line 79, in decorated_main
    return task_function(cfg_passthrough)
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\yolo\v8\segment\train.py", line 153, in train
    model.train(**cfg)
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\yolo\engine\model.py", line 200, in train
    self.trainer.train()
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\yolo\engine\trainer.py", line 185, in train
    self._do_train(int(os.getenv("RANK", -1)), world_size)
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\yolo\engine\trainer.py", line 303, in _do_train
    self.loss, self.loss_items = self.criterion(preds, batch)
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\yolo\v8\segment\train.py", line 45, in criterion
    return self.compute_loss(preds, batch)
  File "C:\Users\****\AppData\Local\Programs\Python\Python310\lib\site-packages\ultralytics\yolo\v8\segment\train.py", line 118, in __call__
    gt_mask = masks[batch_idx == i][mask_idx]
IndexError: The shape of the mask [66, 1] at index 0 does not match the shape of the indexed tensor [16, 160, 160] at index 0
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by mehran66** [comment]
I have the same problem with `overlap_mask=False`

**Comment by Laughing-q** [comment]
@mehran66 hi the issue had been solved in this PR #651 . please make sure you're using the latest version. :)
@saka3324 I noticed you opened a same issue #618 which has been closed, I'm closing this issue also, feel free to reopened it if you have related issues. :)
